external-crates/move/crates/move-compiler/src/cfgir/locals/mod.rs

```diff
@@ -196,19 +271,28 @@ fn command(context: &mut Context, sp!(loc, cmd_): &Command) {
     }
 }
 
-fn lvalues(context: &mut Context, ls: &[LValue]) {
-    ls.iter().for_each(|l| lvalue(context, l))
+fn lvalues(context: &mut Context, case: AssignCase, ls: &[LValue]) {
+    ls.iter().for_each(|l| lvalue(context, case, l))
 }
 
-fn lvalue(context: &mut Context, sp!(loc, l_): &LValue) {
+fn lvalue(context: &mut Context, case: AssignCase, sp!(loc, l_): &LValue) {
     use LValue_ as L;
     match l_ {
         L::Ignore => (),
         L::Var(v, _) => {
+            if case == AssignCase::Update {
+                let mut_ = context.local_mutability(v);
+                if let Some(assign_loc) = context.get_first_assignment(v) {
+                    // If it has already been assigned, it is a mutation.
+                    // This will trigger even if it was assigned and then moved
+                    check_mutability(context, *loc, "assignment", v, mut_, Some(assign_loc));
+                }
+            }
+            context.maybe_set_first_assignment(*v, *loc);
             let ty = context.local_type(v);
             let abilities = ty.value.abilities(ty.loc);
+            let old_state = context.get_state(v);
```

Comment:
super nit: why move this?

Reply:
Ah, yeah I needed it in a previous version of the PR. I'll revert